### PR TITLE
Add support for Swift Package Manager

### PR DESCRIPTION
**Pull request by jbelkins**
### Summary:
- Add SPM-related `.gitignore` entries
- Add a `Package.swift` file that builds the main `Verovio` target plus a `VerovioTests` target for use during development of this PR. (The test target should be deleted before this merges back to the main Verovio repo.)
- Adds the Verovio wrapper class from Shredboard, which serves as a ObjC / Swift interface to access Verovio rendering.
- Within the Verovio wrapper, moved resource setup from a `dispatch_once` block performed before all renders, to being performed on the calling thread immediately prior to each render, due to an internal Verovio change making resources `thread_local`.

Note that Verovio is packaged as a dynamic library, not static, to meet the requirements of the LGPL.  (I verified that when it is integrated into an app, it is indeed embedded as a dynamic framework.)

### References:
https://github.com/rism-digital/verovio/pull/2253 (moves Verovio resources to be `thread_local`)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -95,3 +95,13 @@ Release/
...
 fonts/tmp/
+
+# Default Swift Package Manager .gitignore:
+.DS_Store
+/.build
+/Packages
+/*.xcodeproj
+xcuserdata/
+DerivedData/
+.swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata
```
Comment (jemmons) [inline .gitignore:106]: I think it’s fine to just ignore the whole of `.swiftpm`. That’s what I do with most of my packages.

**Review by jemmons — APPROVED** [review]
Everything looks like it's in the right place to me, but I haven't really scrutinized the compiler directives or anything. If you want me to really dig in, let me know. Otherwise 👍

File: .gitignore (hunk 1 above)
Comment (jbelkins) [reply .gitignore:106]: This is the default `.gitignore` that you get when you start a new project with `swift package init` on latest Xcode.  I figured this must follow the latest guidance.

And I didn't get into the details but a couple discussions on the Swift boards indicated that some things in `.swiftpm` should be kept but not others.  So I figured this should pretty much distill that guidance.

Comment (jemmons) [reply .gitignore:106]: So my understanding is `.swiftpm` is there for tools that work on packages to store configuration. For example, when Xcode opens a package, it puts an autogenerated project file in there. So I guess the question is “Is there any problem checking in an Xcode project file (or some other tool’s equivalent)?”

The answer would seem to be “No.” After all, we check in project files all the time in our other repositories. The only thing I'd caution against is: usually, when we add a project to a repo, that project is the source of all the build settings. In the case of SPM, it is explicitly not. In fact, it's important that nothing in `.swiftpm` have any effect on build behavior. If we accidentally come to rely on behavior that is somehow dependent on configuration living in this directory, builds will start failing when imported into different projects on different hosts and CIs.

So something to keep an eye on.

**Comment by jbelkins** [comment]
@jemmons Please re-review.  This now works with the `ShredboardPreview` app, which is being moved into its own repo.

**Review by jemmons — APPROVED** [review]

**Pull request merged**